File: backend/src/main/java/cohappy/backend/model/dto/request/CreateDebtDTO.java
package cohappy.backend.model.dto.request;

import cohappy.backend.model.DebtType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CreateDebtDTO {
    private String creditorCode;
    private Map<String,Boolean> receiverCode;
    private float amount;
    private String description;
    private DebtType debtType;
}

File: frontend/app/src/main/java/cohappy/frontend/client/dto/request/CreateDebtDTO.kt
package cohappy.frontend.client.dto.request

import cohappy.frontend.client.dto.enum.DebtType
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class CreateDebtDTO(
    val creditorCode: String,
    val receiverCode: Map<String, Boolean>,
    var isCreatorIncluded: Boolean? = null,
    val amount: Float,
    var description: String? = null,
    val debtType: DebtType
)